src/main/java/seedu/address/ui/TodoListPanel.java

```diff
@@ -25,6 +42,65 @@ public TodoListPanel(ObservableList<Task> todoList) {
         super(FXML);
         todoListView.setItems(todoList);
         todoListView.setCellFactory(listView -> new TodoListViewCell());
+        updateCompletionChart(todoList);
+        updateFutureBar(todoList);
+    }
+
+    /**
+     * Updates the completion chart and the description with the required data.
+     *
+     * @param todoList todoList for filling the chart
+     */
+    public void updateCompletionChart(ObservableList<Task> todoList) {
+        int[] statistics = TodoListUtil.getStatistics(todoList);
+        int overdue = statistics[0];
+        int notCompleted = statistics[1];
+        int completed = statistics[2];
+
+        List<PieChart.Data> dataAsList = new ArrayList<>();
+        dataAsList.add(new PieChart.Data("Overdue [" + overdue + "]", overdue));
+        dataAsList.add(new PieChart.Data("Not Completed [" + notCompleted + "]", notCompleted));
+        dataAsList.add(new PieChart.Data("Completed [" + completed + "]", completed));
+
+        ObservableList<PieChart.Data> pieChartData = FXCollections.observableArrayList(dataAsList);
+        description.setText("Hey Buddy! You have " + (notCompleted + overdue) + " unfinished tasks!");
+
+        completionPie.setData(pieChartData);
+    }
```

Comment:
This class is added because the chart is **not** listening to the `TodoList` but to the `pieChartData`, and I'm not sure if there is a way to make `pieChartData` listens directly from `TodoList` without using method to update it manually? Or perhaps there is a way to do it? 😅